packages/medusa/src/services/payment-provider.ts

```diff
@@ -325,6 +390,33 @@ export default class PaymentProviderService extends TransactionBaseService {
     })
   }
 
+  async createPaymentNew(
+    paymentInput: PaymentProviderDataInput
+  ): Promise<Payment> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const { payment_session, currency_code, amount, provider_id } =
+        paymentInput
+
+      const provider = this.retrieveProvider(provider_id)
+      const paymentData = await provider
+        .withTransaction(transactionManager)
+        .getPaymentData(payment_session)
+
+      const paymentRepo = transactionManager.getCustomRepository(
+        this.paymentRepository_
+      )
+
+      const created = paymentRepo.create({
+        provider_id,
+        amount,
+        currency_code,
+        data: paymentData,
+      })
+
+      return paymentRepo.save(created)
```

Comment:
todo: Add missing await
```suggestion
      return await paymentRepo.save(created)
```